### PR TITLE
Adding CycloneDX SBOM

### PR DESCRIPTION
**Pull request by coheigea**

### CODE REVIEW

**Comment by reta** [comment]
@coheigea thanks for this change, I am wondering what it the future plan for it - would we publish the generate SBOM somewhere?

**Comment by coheigea** [comment]
@reta It will be published in Maven Central like this when we release: https://repo1.maven.org/maven2/org/apache/santuario/xmlsec/4.0.0/xmlsec-4.0.0-cyclonedx.json

File: pom.xml [hunk 1]
```diff
@@ -688,6 +688,23 @@
...
                     <version>3.3.2</version>
                 </plugin>
+                <plugin>
+                    <groupId>org.cyclonedx</groupId>
+                    <artifactId>cyclonedx-maven-plugin</artifactId>
+                    <version>2.7.9</version>
+                    <executions>
+                        <execution>
+                            <id>make-bom</id>
+                            <phase>package</phase>
```
Comment (reta) [inline pom.xml:698]: @coheigea it seems like the configuration needs tuning here, the plugin kicks off before the every other module in the build and starts downloading all Apache CXF snapshots first.

```
[INFO] -------------------------< org.apache.cxf:cxf >-------------------------
[INFO] Building Apache CXF 3.6.3-SNAPSHOT                               [1/202]
[INFO]   from pom.xml
[INFO] --------------------------------[ pom ]---------------------------------
[INFO]
[INFO] --- enforcer:3.3.0:enforce (enforce-maven) @ cxf ---
[INFO] Rule 0: org.apache.maven.enforcer.rules.version.RequireJavaVersion passed
[INFO] Rule 1: org.apache.maven.enforcer.rules.version.RequireMavenVersion passed
[INFO] Rule 2: org.apache.maven.enforcer.rules.dependency.BannedDependencies passed
[INFO]
[INFO] --- cyclonedx:2.7.9:makeAggregateBom (make-bom) @ cxf ---

```

Comment (coheigea) [reply pom.xml:698]: OK thanks, I'll take a look

**Pull request merged**